### ISSUE TITLE
Does Intel caffe use async SGD？

### ISSUE THREAD

**Issue by Agoniii** [intel/caffe#205]
When I run resnet50 with batch_size=8k using `lars` on 16 nodes , I found the output of each node is inconsistent. And I only get accuracy(top-1) = 62%.  Does the ASGD affect the accuracy？
Here is my solver.prototxt
```
net: "models/resnet50/train_val_8k.prototxt"

test_iter: 3125
test_interval: 40
test_initialization: false

base_lr: 6.4 # 8k
display: 5
max_iter: 14000
lr_policy: "poly"
power: 2
momentum: 0.9
weight_decay: 0.0001

solver_mode: CPU

snapshot: 500
snapshot_prefix: "models/resnet50/snapshots/resnet50"
snapshot_after_train: false

random_seed: 1
iter_size: 2  # 16nodes*b256*2=8k
warmup_iter: 200
warmup_start_lr: 0.2 # 8k
local_lr_auto: true
```

Here is some log.
```
[5] I0502 16:52:12.812652 13716 solver.cpp:312] Iteration 2110, loss = 3.87861
[5] I0502 16:52:12.813184 13716 solver.cpp:333]     Train net output #0: prob = 3.80451 (* 1 = 3.80451 loss)
[5] I0502 16:52:12.813344 13716 sgd_solver.cpp:215] Iteration 2110, lr = 4.61623
[9] I0502 08:52:36.055213 27707 solver.cpp:312] Iteration 1440, loss = 4.31599
[9] I0502 08:52:36.055503 27707 solver.cpp:333]     Train net output #0: prob = 4.31513 (* 1 = 4.31513 loss)
[9] I0502 08:52:36.055634 27707 sgd_solver.cpp:215] Iteration 1440, lr = 5.15114
[7] I0502 16:52:22.922952 27805 solver.cpp:312] Iteration 2305, loss = 3.57176
[7] I0502 16:52:22.923473 27805 solver.cpp:333]     Train net output #0: prob = 3.60547 (* 1 = 3.60547 loss)
[7] I0502 16:52:22.923604 27805 sgd_solver.cpp:215] Iteration 2305, lr = 4.46606
[15] I0502 08:52:44.336526 17121 solver.cpp:312] Iteration 1395, loss = 4.36874
[15] I0502 08:52:44.337020 17121 solver.cpp:333]     Train net output #0: prob = 4.41702 (* 1 = 4.41702 loss)
[15] I0502 08:52:44.337083 17121 sgd_solver.cpp:215] Iteration 1395, lr = 5.18812
[8] I0502 08:52:02.962663 54230 solver.cpp:312] Iteration 1775, loss = 3.90096
[8] I0502 08:52:02.963088 54230 solver.cpp:333]     Train net output #0: prob = 3.69979 (* 1 = 3.69979 loss)
[8] I0502 08:52:02.963204 54230 sgd_solver.cpp:215] Iteration 1775, lr = 4.88002
```

**Comment by Agoniii** [comment]
@fzou1  I have set `USE_MLSL := 1` in Makefile.config
```
# Intel(r) Machine Learning Scaling Library (uncomment to build
# with MLSL for multi-node training)
 USE_MLSL := 1
```

**Comment by fzou1** [comment]
Okay. May I have your log and command?

**Comment by Agoniii** [comment]
My conmmand:
`mpirun -l -machine host -n 16 ./build/tools/caffe train --solver=models/resnet50/solver_8k.prototxt`

solver_8k.prototxt
```
net: "models/resnet50/train_val_8k.prototxt"

test_iter: 3125
test_interval: 40
test_initialization: false

base_lr: 6.4 # 8k
display: 5
max_iter: 14000
lr_policy: "poly"
power: 2
momentum: 0.9
weight_decay: 0.0001

solver_mode: CPU

snapshot: 500
snapshot_prefix: "models/resnet50/snapshots/resnet50"
snapshot_after_train: false

random_seed: 1
iter_size: 2  # 16nodes*b256*2=8k
warmup_iter: 200
warmup_start_lr: 0.2 # 8k
local_lr_auto: true
```
The log is too long, how to give you ?

**Comment by fzou1** [comment]
Please attach it directly. thanks.

**Comment by Agoniii** [comment]
@fzou1 The file is too big.  I got your email address from your page. Please pardon my presumption in emailing to you.

**Comment by fzou1** [comment]
Solved by removing OpenMPI variable in environment.
